client/src/tests/ProductDetails/pdrender.test.js

```diff
@@ -1,40 +1,209 @@
   });
 
-  it('should render style selector', () => {
-    render(<ProductStyleSelector />);
-    expect(screen.getByText('ProductStyleSelector')).toBeTruthy();
-    //TODO: complete tests
+  it('should render product sale price', () => {
+    render(<ProductPrice selectedStyle={ExampleProductStyles.results[2]} />);
+    expect(screen.getByText(`$${ExampleProductStyles.results[2].sale_price}`)).toBeTruthy();
+    expect(screen.getByText(`$${ExampleProductStyles.results[2].original_price}`)).toBeTruthy();
+  })
 
+  it('should render product info', () => {
+    render(<ProductInfo info={ExampleProductInfo} />);
+    expect(screen.getByText(ExampleProductInfo.slogan)).toBeTruthy(); 
+    expect(screen.getByText(ExampleProductInfo.description)).toBeTruthy(); 
+    ExampleProductInfo.features.forEach(feature => {
+      expect(screen.getByText(`${feature.feature}: ${feature.value}`)).toBeTruthy(); 
+    });
   });
 
-  it('should render product details', () => {
-    render(<ProductDetails />);
-    expect(screen.getByRole('heading', { level: 1 })).toBeTruthy();
-    //TODO: complete tests
+  it('should render share buttons', () => {
+    render(<SocialShare />);
+    const shareButtons = screen.getAllByRole('button')
+    expect(shareButtons.length).toBe(3);
+    fireEvent.click(shareButtons[0]);
+  });
+
+  it('should render select style thumbnails', () => {
+    const setPhotoIndex = jest.fn(); 
+    const photos = ExampleProductStyles.results[0].photos;
+    const styleName = ExampleProductStyles.results[0].name;
+    render(<Thumbnails photos={photos} photoIndex={0} styleName={styleName} setPhotoIndex={setPhotoIndex}  />)
+    photos.forEach((photo, i) => {
+      expect(screen.getByLabelText(`${styleName} preview select ${i + 1} / ${photos?.length}`)).toBeTruthy(); 
+      expect(screen.getByRole('img', {name: `${styleName} preview ${i + 1} / ${photos?.length}`})).toBeTruthy();
+    });
+    const radioOpts = screen.getAllByRole('radio'); 
+    fireEvent.click(radioOpts[2]); 
+    expect(setPhotoIndex).toHaveBeenCalledTimes(2); 
+    const nextButton = screen.getAllByRole('button'); 
+    expect(nextButton.length).toBe(1); 
+    fireEvent.click(nextButton[0]); 
+    
+  });
+
+  it('should render image carousel with only next button', () => {
+    const setExpanded = jest.fn(); 
+    const setPhotoIndex = jest.fn(); 
+    const photos = ExampleProductStyles.results[0].photos;
+    const styleName = ExampleProductStyles.results[0].name;
+    render(<Carousel containerRef={null} photoIndex={0} photos={photos} styleName={styleName} expanded={false} setExpanded={setExpanded} setPhotoIndex={setPhotoIndex}  />);
+    const img = screen.getByRole('img')
+    expect(img).toBeTruthy(); 
+    fireEvent.click(img); 
+    expect(setExpanded).toHaveBeenCalledTimes(1); 
+    const buttons = screen.getAllByRole('button')
+    expect(buttons.length).toBe(1); 
+    fireEvent.click(buttons[0]);
+    expect(setPhotoIndex).toHaveBeenCalledTimes(1); 
```

Comment:
this is kinda what i expected to see after the other clicks in the other tests. to expect some behavior